pallets/parachain-staking/src/lib.rs

```diff
@@ -1937,7 +1938,6 @@ pub mod pallet {
 				<AtStake<T>>::insert(next, account, exposure);
 				Self::deposit_event(Event::CollatorChosen(next, account.clone(), amount));
 			}
-			collators.sort();
```

Comment:
Or is this the core change? Is it enough to not sort by collator id here? I don't see any reason we need this list sorted by collator id.

Reply:
> I don't see any reason we need this list sorted by collator id.

`is_selected_candidate` uses binary search so that's why they're stored sorted by AccountId https://github.com/PureStake/moonbeam/blob/master/pallets/parachain-staking/src/lib.rs#L1658-L1660

The core change is to sort the AccountIds before returning in `compute_top_candidates` instead of here.